### ISSUE TITLE
.any? does not work with both a group and ordering by a column alias

### ISSUE THREAD

**Issue by malmckay** [rails/rails#20508]
Example spec fail here - https://gist.github.com/malmckay/956f7777e1c2ad2281ef

I want a list of author_id, count of author's post and the date the last post was created.

I can make query for this:

```
Post.select("author_id, count(author_id) AS author_count, max(created_at) as lastest_post_date")
    .order("author_count DESC")
    .group("author_id")
```

But, calling `.any?` on that relation will raise.

In `def execute_grouped_calculation`, we could remove .order like so:

```
relation = except(:group, :order)
```

Which might be okay, because counts should not care about order.

**Comment by al2o3cr** [comment]
Grouped counts return results in a particular order, so they *do* care about the order clause. In your example, swapping in `exists?` for `any?` results in a functioning query:
```sql
SELECT  1 AS one FROM "posts" GROUP BY "posts"."author_id" LIMIT 1
```

**Comment by jmbejar** [comment]
@malmckay I was working on fixing this issue and other related problem. Let me know if you have some feedback to this PR https://github.com/rails/rails/issues/20593

**Comment by kamipo** [comment]
This has been fixed since 5.1.
